src/Compilers/Core/Portable/InternalUtilities/InterlockedOperations.cs

```diff
@@ -18,9 +21,9 @@ internal static class InterlockedOperations
         /// <returns>The new value referenced by <paramref name="target"/>. Note that this is
         /// nearly always more useful than the usual return from <see cref="Interlocked.CompareExchange{T}(ref T, T, T)"/>
         /// because it saves another read to <paramref name="target"/>.</returns>
-        public static T Initialize<T>(ref T target, T value) where T : class
+        public static T Initialize<T>(ref T? target, T value) where T : class
         {
-            Debug.Assert((object)value != null);
+            RoslynDebug.Assert((object?)value != null);
```

Comment:
>(object?)value != null [](start = 31, length = 22)

Consider using `value is object` instead.

Reply:
➡️ I'll leave that for a separate code style change.